The job **`aarch64-gnu-llvm-19-1`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/16078266721/job/45378281638) [(plain enhanced)](https://triage.rust-lang.org/gha-logs/rust-lang/rust/45378281638) [(plain)](https://github.com/rust-lang/rust/commit/11220338ae8be0bb6da72db5dbc3efa9c5954ffe/checks/45378281638/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
diff of stderr:

2   --> $DIR/issue-109152.rs:5:21
3    |
4 LL |     vec![42].iter().map(drop);
-    |                     ^^^^----^
+    |                     ---^----^
6    |                     |   |
7    |                     |   this function returns `()`, which is likely not what you wanted
8    |                     |   called `Iterator::map` with callable that returns `()`
---
To only update this specific test, also pass `--test-args lint/issue-109152.rs`

error: 1 errors occurred comparing output.
status: exit status: 1
command: env -u RUSTC_LOG_COLOR RUSTC_ICE="0" RUST_BACKTRACE="short" "/checkout/obj/build/aarch64-unknown-linux-gnu/stage2/bin/rustc" "/checkout/tests/ui/lint/issue-109152.rs" "-Zthreads=1" "-Zsimulate-remapped-rust-src-base=/rustc/FAKE_PREFIX" "-Ztranslate-remapped-path-to-local-path=no" "-Z" "ignore-directory-in-diagnostics-source-blocks=/cargo" "-Z" "ignore-directory-in-diagnostics-source-blocks=/checkout/vendor" "--sysroot" "/checkout/obj/build/aarch64-unknown-linux-gnu/stage2" "--target=aarch64-unknown-linux-gnu" "--check-cfg" "cfg(test,FALSE)" "--error-format" "json" "--json" "future-incompat" "-Ccodegen-units=1" "-Zui-testing" "-Zdeduplicate-diagnostics=no" "-Zwrite-long-types-to-disk=no" "-Cstrip=debuginfo" "--emit" "metadata" "-C" "prefer-dynamic" "--out-dir" "/checkout/obj/build/aarch64-unknown-linux-gnu/test/ui/lint/issue-109152" "-A" "unused" "-A" "internal_features" "-Crpath" "-Cdebuginfo=0" "-Lnative=/checkout/obj/build/aarch64-unknown-linux-gnu/native/rust-test-helpers"
stdout: none
--- stderr -------------------------------
error: `Iterator::map` call that discard the iterator's values
##[error]  --> /checkout/tests/ui/lint/issue-109152.rs:5:21
   |
LL |     vec![42].iter().map(drop);
   |                     ---^----^
   |                     |   |
   |                     |   this function returns `()`, which is likely not what you wanted
   |                     |   called `Iterator::map` with callable that returns `()`
   |                     after this call to map, the resulting iterator is `impl Iterator<Item = ()>`, which means the only information carried by the iterator is the number of items
   |
   = note: `Iterator::map`, like many of the methods on `Iterator`, gets executed lazily, meaning that its effects won't be visible until it is iterated
note: the lint level is defined here
  --> /checkout/tests/ui/lint/issue-109152.rs:1:9
   |
LL | #![deny(map_unit_fn)]
   |         ^^^^^^^^^^^
help: you might have meant to use `Iterator::for_each`
   |
LL -     vec![42].iter().map(drop);
LL +     vec![42].iter().for_each(drop);
   |

error: aborting due to 1 previous error
------------------------------------------


---- [ui] tests/ui/lint/lint_map_unit_fn.rs stdout ----
Saved the actual stderr to `/checkout/obj/build/aarch64-unknown-linux-gnu/test/ui/lint/lint_map_unit_fn/lint_map_unit_fn.stderr`
diff of stderr:

5    | --------------------------- this function returns `()`, which is likely not what you wanted
6 ...
7 LL |     x.iter_mut().map(foo);
-    |                  ^^^^---^
+    |                  ---^---^
9    |                  |   |
10    |                  |   called `Iterator::map` with callable that returns `()`
11    |                  after this call to map, the resulting iterator is `impl Iterator<Item = ()>`, which means the only information carried by the iterator is the number of items

25 error: `Iterator::map` call that discard the iterator's values
26   --> $DIR/lint_map_unit_fn.rs:11:18
27    |
- LL |         x.iter_mut().map(|items| {
-    |                      ^   -------
-    |                      |   |
-    |  ____________________|___this function returns `()`, which is likely not what you wanted
-    | |  __________________|
-    | | |
- LL | | |
- LL | | |         items.sort();
- LL | | |     });
-    | | |     -^ after this call to map, the resulting iterator is `impl Iterator<Item = ()>`, which means the only information carried by the iterator is the number of items
-    | | |_____||
-    | |_______|
-    |         called `Iterator::map` with callable that returns `()`
+ LL |        x.iter_mut().map(|items| {
+    |                     ^-- -------
+    |                     |   |
+    |  ___________________|___this function returns `()`, which is likely not what you wanted
+    | |                   |
+    | | __________________after this call to map, the resulting iterator is `impl Iterator<Item = ()>`, which means the only information carried by the iterator is the number of items
+    | ||
+ LL | ||
+ LL | ||         items.sort();
+ LL | ||     });
+    | ||_____-^
+    | |______|
+    |        called `Iterator::map` with callable that returns `()`
41    |
42    = note: `Iterator::map`, like many of the methods on `Iterator`, gets executed lazily, meaning that its effects won't be visible until it is iterated
43 help: you might have meant to use `Iterator::for_each`

53    |             --------------------- this function returns `()`, which is likely not what you wanted
54 ...
55 LL |     x.iter_mut().map(f);
-    |                  ^^^^-^
+    |                  ---^-^
57    |                  |   |
58    |                  |   called `Iterator::map` with callable that returns `()`
59    |                  after this call to map, the resulting iterator is `impl Iterator<Item = ()>`, which means the only information carried by the iterator is the number of items

Note: some mismatched output was normalized before being compared
- LL | ||     //~^ ERROR `Iterator::map` call that discard the iterator's values
+    |                  ---^---^
+ LL |        x.iter_mut().map(|items| {
+    |                     ^-- -------
+    |                     |   |
+    |  ___________________|___this function returns `()`, which is likely not what you wanted
+    | |                   |
+    | | __________________after this call to map, the resulting iterator is `impl Iterator<Item = ()>`, which means the only information carried by the iterator is the number of items
+    | ||
+ LL | ||
+ LL | ||         items.sort();
+ LL | ||     });
+    | ||_____-^
---
To only update this specific test, also pass `--test-args lint/lint_map_unit_fn.rs`

error: 1 errors occurred comparing output.
status: exit status: 1
command: env -u RUSTC_LOG_COLOR RUSTC_ICE="0" RUST_BACKTRACE="short" "/checkout/obj/build/aarch64-unknown-linux-gnu/stage2/bin/rustc" "/checkout/tests/ui/lint/lint_map_unit_fn.rs" "-Zthreads=1" "-Zsimulate-remapped-rust-src-base=/rustc/FAKE_PREFIX" "-Ztranslate-remapped-path-to-local-path=no" "-Z" "ignore-directory-in-diagnostics-source-blocks=/cargo" "-Z" "ignore-directory-in-diagnostics-source-blocks=/checkout/vendor" "--sysroot" "/checkout/obj/build/aarch64-unknown-linux-gnu/stage2" "--target=aarch64-unknown-linux-gnu" "--check-cfg" "cfg(test,FALSE)" "--error-format" "json" "--json" "future-incompat" "-Ccodegen-units=1" "-Zui-testing" "-Zdeduplicate-diagnostics=no" "-Zwrite-long-types-to-disk=no" "-Cstrip=debuginfo" "--emit" "metadata" "-C" "prefer-dynamic" "--out-dir" "/checkout/obj/build/aarch64-unknown-linux-gnu/test/ui/lint/lint_map_unit_fn" "-A" "unused" "-A" "internal_features" "-Crpath" "-Cdebuginfo=0" "-Lnative=/checkout/obj/build/aarch64-unknown-linux-gnu/native/rust-test-helpers"
stdout: none
--- stderr -------------------------------
error: `Iterator::map` call that discard the iterator's values
##[error]  --> /checkout/tests/ui/lint/lint_map_unit_fn.rs:9:18
   |
LL | fn foo(items: &mut Vec<u8>) {
   | --------------------------- this function returns `()`, which is likely not what you wanted
...
LL |     x.iter_mut().map(foo);
   |                  ---^---^
   |                  |   |
   |                  |   called `Iterator::map` with callable that returns `()`
   |                  after this call to map, the resulting iterator is `impl Iterator<Item = ()>`, which means the only information carried by the iterator is the number of items
   |
   = note: `Iterator::map`, like many of the methods on `Iterator`, gets executed lazily, meaning that its effects won't be visible until it is iterated
note: the lint level is defined here
  --> /checkout/tests/ui/lint/lint_map_unit_fn.rs:1:9
   |
LL | #![deny(map_unit_fn)]
   |         ^^^^^^^^^^^
help: you might have meant to use `Iterator::for_each`
   |
LL -     x.iter_mut().map(foo);
LL +     x.iter_mut().for_each(foo);
   |

error: `Iterator::map` call that discard the iterator's values
##[error]  --> /checkout/tests/ui/lint/lint_map_unit_fn.rs:11:18
   |
LL |        x.iter_mut().map(|items| {
   |                     ^-- -------
   |                     |   |
   |  ___________________|___this function returns `()`, which is likely not what you wanted
   | |                   |
   | | __________________after this call to map, the resulting iterator is `impl Iterator<Item = ()>`, which means the only information carried by the iterator is the number of items
   | ||
LL | ||     //~^ ERROR `Iterator::map` call that discard the iterator's values
LL | ||         items.sort();
LL | ||     });
   | ||_____-^
   | |______|
   |        called `Iterator::map` with callable that returns `()`
   |
   = note: `Iterator::map`, like many of the methods on `Iterator`, gets executed lazily, meaning that its effects won't be visible until it is iterated
help: you might have meant to use `Iterator::for_each`
   |
LL -     x.iter_mut().map(|items| {
LL +     x.iter_mut().for_each(|items| {
   |

error: `Iterator::map` call that discard the iterator's values
##[error]  --> /checkout/tests/ui/lint/lint_map_unit_fn.rs:18:18
   |
LL |     let f = |items: &mut Vec<u8>| {
   |             --------------------- this function returns `()`, which is likely not what you wanted
...
LL |     x.iter_mut().map(f);
   |                  ---^-^
   |                  |   |
   |                  |   called `Iterator::map` with callable that returns `()`
   |                  after this call to map, the resulting iterator is `impl Iterator<Item = ()>`, which means the only information carried by the iterator is the number of items
   |
   = note: `Iterator::map`, like many of the methods on `Iterator`, gets executed lazily, meaning that its effects won't be visible until it is iterated
help: you might have meant to use `Iterator::for_each`
   |
LL -     x.iter_mut().map(f);
LL +     x.iter_mut().for_each(f);
   |

error: aborting due to 3 previous errors
------------------------------------------

```

</details>

--- a/compiler/rustc_const_eval/src/const_eval/mod.rs
+++ b/compiler/rustc_const_eval/src/const_eval/mod.rs
@@ -1,9 +1,9 @@
 // Not in interpret to make sure we do not use private implementation details
 
 use rustc_abi::{FieldIdx, VariantIdx};
-use rustc_middle::query::Key;
 use rustc_middle::ty::{self, Ty, TyCtxt};
 use rustc_middle::{bug, mir};
+use rustc_span::DUMMY_SP;
 use tracing::instrument;
 
 use crate::interpret::InterpCx;
@@ -71,8 +71,7 @@ pub fn tag_for_variant_provider<'tcx>(
     let (ty, variant_index) = key.value;
     assert!(ty.is_enum());
 
-    let ecx =
-        InterpCx::new(tcx, ty.default_span(tcx), key.typing_env, crate::const_eval::DummyMachine);
+    let ecx = InterpCx::new(tcx, DUMMY_SP, key.typing_env, crate::const_eval::DummyMachine);
 
     let layout = ecx.layout_of(ty).unwrap();
     ecx.tag_for_variant(layout, variant_index).unwrap().map(|(tag, _tag_field)| tag)

--- a/compiler/rustc_lint/src/map_unit_fn.rs
+++ b/compiler/rustc_lint/src/map_unit_fn.rs
@@ -1,5 +1,4 @@
 use rustc_hir::{Expr, ExprKind, HirId, Stmt, StmtKind};
-use rustc_middle::query::Key;
 use rustc_middle::ty::{self, Ty};
 use rustc_session::{declare_lint, declare_lint_pass};
 
@@ -69,7 +68,7 @@ impl<'tcx> LateLintPass<'tcx> for MapUnitFn {
                                         .span_of_impl(*id)
                                         .unwrap_or(default_span),
                                     argument_label: args[0].span,
-                                    map_label: arg_ty.default_span(cx.tcx),
+                                    map_label: span,
                                     suggestion: path.ident.span,
                                     replace: "for_each".to_string(),
                                 },
@@ -88,7 +87,7 @@ impl<'tcx> LateLintPass<'tcx> for MapUnitFn {
                                         .span_of_impl(*id)
                                         .unwrap_or(default_span),
                                     argument_label: args[0].span,
-                                    map_label: arg_ty.default_span(cx.tcx),
+                                    map_label: span,
                                     suggestion: path.ident.span,
                                     replace: "for_each".to_string(),
                                 },

--- a/tests/ui/lint/lint_map_unit_fn.stderr
+++ b/tests/ui/lint/lint_map_unit_fn.stderr
@@ -25,19 +25,18 @@ LL +     x.iter_mut().for_each(foo);
 error: `Iterator::map` call that discard the iterator's values
   --> $DIR/lint_map_unit_fn.rs:11:18
    |
-LL |         x.iter_mut().map(|items| {
-   |                      ^   -------
-   |                      |   |
-   |  ____________________|___this function returns `()`, which is likely not what you wanted
-   | |  __________________|
-   | | |
-LL | | |
-LL | | |         items.sort();
-LL | | |     });
-   | | |     -^ after this call to map, the resulting iterator is `impl Iterator<Item = ()>`, which means the only information carried by the iterator is the number of items
-   | | |_____||
-   | |_______|
-   |         called `Iterator::map` with callable that returns `()`
+LL |        x.iter_mut().map(|items| {
+   |                     ^   -------
+   |                     |   |
+   |  ___________________|___this function returns `()`, which is likely not what you wanted
+   | | __________________|
+   | ||
+LL | ||
+LL | ||         items.sort();
+LL | ||     });
+   | ||_____-^ after this call to map, the resulting iterator is `impl Iterator<Item = ()>`, which means the only information carried by the iterator is the number of items
+   | |______|
+   |        called `Iterator::map` with callable that returns `()`
    |
    = note: `Iterator::map`, like many of the methods on `Iterator`, gets executed lazily, meaning that its effects won't be visible until it is iterated
 help: you might have meant to use `Iterator::for_each`

